src/api/services/adminServiceHelpers.ts

```diff
@@ -93,6 +93,11 @@ export type ApiKeyDTO = {
   service_id: number;
 };
 
+export type CreatedApiKeyDTO = {
+  authorizable: ApiKeyDTO & { key_hash: string; key_salt: string; secret: string };
```

Comment:
Is there a nice way we can do this which avoid the `& { ... }`. I know the object we get from admin is constructed weirdly, but perhaps we could just manipulate it ourselves so that it is something like:
```
export type CreatedApiKeyDTO = {
  apiKey: ApiKeyDTO
  secrets: ApiKeySecretsDTO;
};
```
Happy to be convinced otherwise.

Reply:
I'm not sure how that would be done. That's the type of return we get from admin. The issue is that it's adding fields inside authorizable and then adding plaintext_key as well. We could alter it once we recieve but we'd still need this object I think.
```
{
  "authorizable": {
    "key_hash": "Sk7OaQC7AUyZTeB1ubp8iE1FgL8QqCaN+pLN3lYgaRaU012qlsc4CXpXUE0CJxep8f8xtvFzjiBlad69saKyRA==",
    "key_salt": "aUFdfV+WYov+CG2Y/xdztBIbqlT+n3pfyEBfHdhwIUc=",
    "secret": "6Le7vmRaZMlPzRldVuFbCU9Nf44nL6uPnX69VglWVpU=",
    "name": "TESTING",
    "contact": "Special_UID2-C-L-999-ZY967",
    "created": 1704671901,
    "roles": [
      "MAPPER"
    ],
    "site_id": 999,
    "disabled": false,
    "service_id": 0,
    "key_id": "UID2-C-L-999-ZY967"
  },
  "plaintext_key": "UID2-C-L-999-ZY967t.aZ0rPhS347aULe2qraDOpvKYWKj6pdvUfw1jQ="
}
```